src/Attributes/Column.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace PDGA\DataObjects\Attributes;
+
+use Attribute;
+
+#[Attribute(Attribute::TARGET_PROPERTY)]
+class Column
+{
+    private string $name;
+    private string $sqlDataType;
+    private int $maxLength;
+    private bool $isPrimary;
+    private bool $isGenerated;
+    private bool $hasDefault;
+
+    /**
+     * Constructor for Column attribute.
+     *
+     * @param string $name - The name of the column. Required.
+     * @param string $sqlDataType - The datatype of the column in the database. Defaults to null.
+     * @param int $maxLength - The maximum length of the column. Defaults to null.
+     * @param bool $isPrimary - Boolean to indicate if the column is a primary column. Defaults to false.
+     * @param bool $isGenerated - Boolean to indicate if the column is auto-generated. Defaults to false.
+     * @param bool $hasDefault - Boolean to indicate if the column has a default value. Defaults to false.
+     */
+    public function __construct(
+        string $name,
+        string $sqlDataType = null,
+        int $maxLength = null,
+        bool $isPrimary = false,
+        bool $isGenerated = false,
+        bool $hasDefault = false
+    )

```

Comment:
You could use property promotion on these to shorten this code by prefixing each parameter with `private`, getting rid of the explicit declaration of the members (Lines 10-16), and getting rid of the body of the ctor.